atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/async/CqlClient.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.keyvalue.cassandra.async;
+
+import java.io.Closeable;
+import java.util.List;
+import java.util.function.Function;
+import java.util.stream.Stream;
+
+import com.datastax.driver.core.BoundStatement;
+import com.datastax.driver.core.PreparedStatement;
+import com.datastax.driver.core.ResultSet;
+import com.datastax.driver.core.Row;
+import com.google.common.util.concurrent.ListenableFuture;
+
+public interface CqlClient extends Closeable {
+
+    interface CqlQuery<V, R> {
+        BoundStatement boundStatement();
+
+        Visitor<V, R> createVisitor();
+    }
+
+    interface Visitor<V, R> {
+
+        void visit(V value);
```

Comment:
I'm not really sure what the difference between `V value`, `R result` and `Row row` is, plus this leads to a somewhat imperative api where (assuming I'm collecting this into a set or something) I have to make an empty collection, and then one by one add to that collection. Or as another example, I have to assign to a variable/mutable field somewhere that's outside the visitor. I would really like to see how this api is supposed to be used by having the health check use this api.